src/topotoolbox/stream_object.py

```diff
@@ -0,0 +1,78 @@
+"""This module contains the StreamObject class.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .flow_object import FlowObject
+
+# pylint: disable=no-name-in-module
+from . import _flow  # type: ignore
+
+_all_ = ['StreamObject']
+
+
+class StreamObject():
+    """A class to represent stream flow accumulation based on a FlowObject.
+    """
+
+    def __init__(self, flow: FlowObject) -> None:
+
+        acc = np.zeros_like(flow.target, order='F', dtype=np.float32)
+        weights = np.ones_like(flow.target, order='F', dtype=np.float32)
+
+        _flow.flow_accumulation(
+            acc, flow.source, flow.direction, weights, flow.shape)
+
+        self.path = flow.path
+        self.name = flow.name
+
+        # raster metadata
+        self.acc = acc
+        self.shape = flow.shape
+
+        # georeference
+        self.bounds = flow.bounds
+        self.transform = flow.transform
+        self.crs = flow.crs
+
+    def show(self, cmap='binary') -> None:
+        """
+        Display the StreamObject instance as an image using Matplotlib.
+
+        Parameters
+        ----------
+        cmap : str, optional
+            Matplotlib colormap that will be used in the plot.
+        """
+        plt.imshow(self.acc, cmap=cmap)
+        plt.title(self.name)
+        plt.colorbar()
+        plt.tight_layout()
+        plt.show()
```

Comment:
It is probably fine for now to plot the accumulation or the logical `GridObject` defining the stream network (see the other comments).

The MATLAB version plots the stream network as a collection of lines. This is somewhat complicated because it requires splitting off each branch of the stream network and plotting it independently, which in turn requires traversing the graph of the stream network. It might be best to save that for future work.